### PR TITLE
Add Requests Manager to View All Prayers, Centcom/Syndicate Requests & Nuke Code Requests in Round

### PR DESCRIPTION
**Pull request by bobbah**
## About The Pull Request

Implements the Requests Manager, a new admin tool used to view all prayers, Centcom/Syndicate requests, and nuke code requests within the span of a round.

requests_example_5

The verb can be found under ``Admin.Game``, or alternatively just use the verb directly ``Requests Manager``. The requests can be filtered by type (using the type filter button), or using a text search which searches both the message text as well as the request owner's name.

[Fulfills the request of WineAllWine](https://tgstation13.org/phpBB/viewtopic.php?f=5&t=29679)

## Why It's Good For The Game

Gives admins the ability to see a very easy to use recap of all requests that occur in a round. This system is readily extendible to support other types of requests, should someone care to add it.

## Technical Note for Future Readers

This was originally intended to have prefs added such that you could include sound notifications for each type of request. This was regrettably removed from this PR due to ongoing changes to the codebase; however, I would highly recommend re-visiting my implementation [found on the branch here](https://github.com/bobbahbrown/tgstation/tree/f79b36c7f1ab82a7f915702aee368a4954b55ed2), at that specific commit, prior to being removed from the branch with further commits.

## Changelog
:cl: bobbahbrown
admin: Add Requests Manager to view all prayers/centcom and syndicate requests/and nuke code requests within a round.
/:cl:

### CODE REVIEW

**Comment by bobbahbrown** [comment]
Updated to change the color of centcom/nuke requests to yellow, as it looks better, and added two means of filtering: a type panel, as well as a text search which searches the request message as well as the owner's name.

**Comment by GoldenAlpharex** [comment]
Very sexy, I'm really hype for it.

File: tgui/packages/tgui/interfaces/RequestManager.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
+/**
+ * @file
+ * @copyright 2021 bobbahbrown (https://github.com/bobbahbrown)
+ * @license MIT
+ */
+
+import { Button, Section, Table, Input } from '../components';
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/RequestManager.js:7]: Alt+Shift+O

File: tgui/packages/tgui/interfaces/RequestManager.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      (request) =>
+        decodeHtmlEntities(request.message)
+          .toLowerCase()
+          .includes(filterText)
+        || request.owner_name.toLowerCase().includes(filterText)
+    );
+  }
+
+  return (
+    <Window width={575} height={600}>
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/RequestManager.js:41]: Title would be nice

File: tgui/packages/tgui/styles/interfaces/RequestManager.scss [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+.RequestManager__row:hover .RequestManager__controlsContainer {
+  opacity: 1;
+}
+
+.RequestManager__controlsContainer {
+  opacity: 0;
+  position: absolute;
+  right: 0;
+  top: 0;
```
Comment (stylemistake) [inline tgui/packages/tgui/styles/interfaces/RequestManager.scss:89]: 3px would probably work better for top, buttons just look visually offset.

Comment (bobbahbrown) [reply tgui/packages/tgui/styles/interfaces/RequestManager.scss]: I agree, added to right as well as it looked better.

**Comment by JoshAdamPowell** [comment]
As the patron of this PR I'm extremely happy with it and I've made a donation to the British Royal legion. I will try and see if my work's charity budget will be able to match it.

**Review by stylemistake — APPROVED** [review]

**Comment by stylemistake** [comment]
I see big potential in this kind of UIs for admins, amazing work.

**Comment by Crossedfall** [comment]
Just needs a "Smite All" button and it's perfect.

**Review by Timberpoes — CHANGES_REQUESTED** [review]
Your ui_act never returns TRUE when you've successfully handled a user action. I'm not sure if this is still a requirement but last time I was coding tgui not returning TRUE when you've handled a user input meant interfaces didn't update properly. 

You've got some error states where you just silently return - For example, missing R_FUN when trying to smite or replying to prayers.

Can you add some to_chat feedback to any remaining `if(some_problem) return` checks to let the admin know why the button they pushed failed?

**Comment by stylemistake** [comment]
Yes, what timber said, I have missed this detail during review.

File: code/modules/admin/verbs/pray.dm [hunk 4]
```diff
@@ -40,22 +40,20 @@
...
 	to_chat(usr, span_info("You pray to the gods: \"[msg_tmp]\""), confidential = TRUE)
 
 	SSblackbox.record_feedback("tally", "admin_verb", 1, "Prayer") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
-	//log_admin("HELP: [key_name(src)]: [msg]")
+
 
 /// Used by communications consoles to message CentCom
 /proc/message_centcom(text, mob/sender)
 	var/msg = copytext_char(sanitize(text), 1, MAX_MESSAGE_LEN)
+	GLOB.requests.message_centcom(usr.client, msg)
```
Comment (ZephyrTFA) [inline code/modules/admin/verbs/pray.dm:56]:
```suggestion
	GLOB.requests.message_centcom(sender.client, msg)
```
You already have a reference to the sender

**Comment by bobbahbrown** [comment]
sorry for the delay on the updates, i didn't notice the comments until much later than i should have!

**Review by Timberpoes — CHANGES_REQUESTED** [review]

**Review by Timberpoes — CHANGES_REQUESTED** [review]

**Review by Timberpoes — APPROVED** [review]
Stylemistake was happy with the other js-y stuff so I think this is good to go now.

**Pull request merged**